Import the original AB-3 LearningOutcomes

pyokagan opened the pull request:
As we are planning to replace AB-3 with this repo, let's import the
original AB-3 Learning Outcomes document and diagrams from the AB-3 repo
(as of commit 841298aa618dbffe710469639ed922c27752cd36).

For now, we will import the document as-is, even though the source code
in this repo has some differences compared to the original AB-3, and so
some of the Learning Outcomes exercises can't be directly applied. We
will update the Learning Outcomes document incrementally in later PRs.

The original authors of the Learning Outcomes document are listed below
(in alphabetical order):

Co-authored-by: Damith C. Rajapakse <damith@gmail.com>
Co-authored-by: Jun An <E0002937@u.nus.edu>
Co-authored-by: Paul Tan <pyokagan@gmail.com>
Co-authored-by: Tan Wang Leng <tanwangleng@outlook.com>

damithc commented:
How does the new GUI look?

damithc commented on src/seedu/addressbook/ui/DarkTheme.css:
```diff
@@ -0,0 +1,15 @@
+.text-area {
+    -fx-background-color: black;
+    -fx-control-inner-background: black;
+    -fx-font-size: 12px;
```
Why use `pt` above but `px` here? In the screenshot, the two text sizes seem different.

damithc commented:
@yl-coder is this one done. Merge if yes?

yl-coder commented:
px is appear differently on different screen. So i guess i will change to pt for consistency.

pyokagan committed:
Import the original AB-3 LearningOutcomes

As we are planning to replace AB-3 with this repo, let's import the
original AB-3 Learning Outcomes document and diagrams from the AB-3 repo
(as of commit 841298aa618dbffe710469639ed922c27752cd36).

For now, we will import the document as-is, even though the source code
in this repo has some differences compared to the original AB-3, and so
some of the Learning Outcomes exercises can't be directly applied. We
will update the Learning Outcomes document incrementally in later PRs.

The original authors of the Learning Outcomes document are listed below
(in alphabetical order):

Co-authored-by: Damith C. Rajapakse <damith@gmail.com>
Co-authored-by: Jun An <E0002937@u.nus.edu>
Co-authored-by: Paul Tan <pyokagan@gmail.com>
Co-authored-by: Tan Wang Leng <tanwangleng@outlook.com>
